options/options_helper.cc

```diff
@@ -255,6 +263,77 @@ bool ParseVectorCompressionType(
   return true;
 }
 
+// This is to handle backward compatibility, where compaction_options_fifo
+// could be assigned a single scalar value, say, like "23", which would be
+// assigned to max_table_files_size.
+bool FIFOCompactionOptionsSpecialCase(const std::string& opt_str,
+                                      CompactionOptionsFIFO* options) {
+  if (opt_str.find("=") != std::string::npos) {
+    // New format. Go do your new parsing using ParseStructOptions.
+    return false;
+  }
+
+  // Old format. Parse just a single uint64_t value.
+  options->max_table_files_size = ParseUint64(opt_str);
+  return true;
+}
+
+template <typename T>
+bool SerializeStruct(
```

Comment:
can we have one SerializeStruct function that optionally surrounds with curly braces? it'd correspond directly to ParseStructOptions, then. I spent a bit of time trying to understand the relationship between the functions, maybe reducing the number of them if possible would be helpful.

Reply:
Ah, true, that would be much cleaner. Not sure why I didn't think about it. I'll make the change in the next iteration that I currently working on.